### PR TITLE
Fix usage of JS Object property names in data.

### PR DESCRIPTION
**Pull request by davidlehn**
Fixes https://github.com/digitalbazaar/jsonld.js/issues/304.

### CODE REVIEW

File: lib/context.js [hunk 1]
```diff
@@ -686,10 +690,12 @@ api.getInitialContext = options => {
     // create term selections for each mapping in the context, ordered by
     // shortest and then lexicographically least
     const mappings = activeCtx.mappings;
-    const terms = Object.keys(mappings).sort(_compareShortestLeast);
-    for(let i = 0; i < terms.length; ++i) {
-      const term = terms[i];
-      const mapping = mappings[term];
+    const terms = [...mappings.keys()].sort(_compareShortestLeast);
+    for(const term of terms) {
```
Comment (mattcollier) [inline lib/context.js]: would it be more performance to just do `get` first and then check for `undefined` or `null` or whatever?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+- Usage of JavaScript Object property names in data.
+  - **NOTE**: A class of bugs was causing term names such as `toString`,
+    `valueOf`, and others to be dropped or produce bogus output. The fix could
+    cause output triples to differ from previous versions if those special
+    names were used.
+  - Specifically, the problem was using `x in obj` instead of
+    `obj.hasOwnProperty(x)` or a `Map`.
+  - Fixed usage in contexts for expand and compact.
+  - Attempted fixes in other parts of the code with similar `x in obj` usage.
+    Finding actual realistic failing test cases proved difficult.
```
Comment (mattcollier) [inline CHANGELOG.md]: I'm not sure this belongs in the changelog, maybe create an issue if positive/negative tests are needed.

File: lib/context.js (hunk 1 above)
Comment (davidlehn) [reply lib/context.js]: I haven't tested the difference, but it probably is faster when that pattern is ok.  I didn't consider it enough.  I don't think `undefined` is a value so perhaps that could be used.

Might also be faster to sort `mapping.entries()` and iterate through `[key, mapping]` pairs to just avoid `has` and `get` calls.

**Comment by dlongley** [comment]
@davidlehn -- apologies, this has conflicts now. It will need to update the `@protected` stuff to use a `Map`.

**Pull request merged**